common/simulation/spi.cpp

```diff
@@ -22,11 +22,16 @@ void sim_spi::SimTMC2130Spi::transmit_receive(
     uint8_t reg = control & ~write_mask;
 
     if (control & write_mask) {
+        // This is a write command.
         register_map[reg] = data;
+    } else {
+        // A read command will return the data from the previous read command.
+        auto out_iter = receive.begin();
+        out_iter =
+            bit_utils::int_to_bytes(read_register, out_iter, receive.end());
+        out_iter = bit_utils::int_to_bytes(register_map[read_register],
+                                           out_iter, receive.end());
+        // For the next read.
+        read_register = reg;
```

Comment:
Can you explain what the purpose of modifying the read register is here?

Reply:
We use `read_register` for the read on line 32 rather than `reg`. `read_register` caches the current request's register for the next read request.